calico/felix/fetcd.py

```diff
@@ -132,6 +149,67 @@ def _periodically_resync(self):
             gevent.sleep(sleep_time)
             self.force_resync(reason="periodic resync", async=True)
 
+    @logging_exceptions
+    def _status_reporting(self):
+        """
+        Greenlet: periodically writes to etcd
+
+        :return: Does not return, unless reporting disabled.
+        """
+        _log.info("Started status reporting thread.")
+        ttl = self._config.REPORTING_TTL_SECS
+        interval = self._config.REPORTING_INTERVAL_SECS
+        _log.debug("Reporting interval: %s, TTL: %s", interval, ttl)
+
+        if interval == 0:
+            _log.info("Interval is 0, status reporting disabled.")
+            return
+
+        while True:
+            try:
+                self.update_felix_status(ttl)
+                gevent.sleep(interval)
+            except EtcdException as e:
+                _log.debug("Status updating failed. %r. Reconnecting... ", e)
+                _reconnect(self)
+                gevent.sleep(RETRY_DELAY)
+
+    def update_felix_status(self, ttl):
+        """
+        Writes two keys to etcd:
+        uptime in secs
+        felix status in JSON - containing current time in ISO 8601 Zulu format
+
+        :param: ttl int: time to live in sec - lifetime of the status report
+        """
+        status_key = key_for_status(self._config.HOSTNAME)
+        status = {}
+
+        time_now = datetime.datetime.fromtimestamp(time.time())
+        time_formatted = time_now.replace(microsecond=0).isoformat()+'Z'
+        status['status_time'] = time_formatted
+
+        status_value = json.dumps(status)
+
+        uptime_key = key_for_uptime(self._config.HOSTNAME)
+        uptime_value = int(time.time() - self._start_time)
+
+        self.write_to_etcd(status_key, status_value, async=True)
```

Comment:
I'm not sure we're getting any benefit from passing this back to the actor as a message. The etcd client is already threadsafe so we should just be abel to use it as `self.client.xxx`